opencog/persist/file/fast_load.cc

```diff
@@ -231,17 +213,37 @@ void opencog::load_file(std::string fname, AtomSpace& as)
             // Trim away comments at end of line
             if (0 < pcount)
             {
-                expr = expr.substr(l, r-l);
+                expr = expr.substr(l, r - l);
                 break;
             }
 
             // Nothing to do.
-            if (l == r) break;
+            if (l == r)
+                break;
 
             expr_cnt++;
-            as.add_atom(recursive_parse(expr, l, r, line_cnt));
-            expr = expr.substr(r+1);
+            h = as.add_atom(recursive_parse(expr, l, r, line_cnt));
+            expr = expr.substr(r + 1);
         }
     }
+
+    return h;
```

Comment:
This is .. kind of strange ... it returns only the last atom in the stream. Is this actually useful? 
Maybe you want to return a `std::vector<Handle>` of them?

Reply:
Oh, I guess its useful if you want to parse only one atom at a time ...

Reply:
> it returns only the last atom in the stream. Is this actually useful?

My initial use case was to return a single atom that wraps the other atoms as the addition to the atomspace is done recursively (using the `recursive_parse` function). But  your suggestion of returning `std::vector<Handle> ` makes more sense.

Reply:
But I am wondering if we should use `std::set<Handle>` instead of a vector to avoid adding the same atom more than once as it can occur multiple times in an input. What do you think?

Reply:
Well, don't over-engineer things. Right now, fast-file-load is fast because it spends about half its time parsing strings, and half it's time in `AtomSpace::add_atom()`. You want to avoid doing things that will slow it down for no good reason.  So:
* Creating giant `SetLink` of atoms is pointless and useless and slow. It takes a lot of CPU time to validate each, a lot of CPU time to set up the incoming sets, and then, if/when you delete the `SetLink`, you waste a lot of time changing the incoming sets back.  It's just a bad idea.
* Maintaining a HandleSeq or a HandleSet is useless if no one is going to look at it. It might not sound like a lot of CPU time, but compared to parsing strings, or compared to `AtomSpace::add_atom()`. it might be another 5% or 10% ... do you really need to slow down the code?
* The ***whole idea*** of the atomspace is that it is a "blackboard", a "place where you can find things". You shouldn't really need to know what was loaded, because you can always find it.  If you feel that you do need to know ... ask yourself why? is it really needed? or is it just a comfortable old habit that has no meaning, has no purpose? (old meaningless habits are great for culture and society, but not for programming.)  I suspect that you don't *really* need the list of loaded atoms.

Reply:
> I suspect that you don't really need the list of loaded atoms.

I don't (at least for now). I was just suggesting solutions to your comment. But we can change this to return a container type when the need arises.

Reply:
Well, do you need the return value at all? Or are you planning for some uncertain future?